tests/test_query.py

```diff
@@ -170,26 +269,103 @@ def test_docker_images(self):
+        product_with_no_helm_charts = InstalledProductVersion(
+            'sat', '0.9.9', {'component_versions': {'helm': {}}}
+        )
+        self.assertEqual(product_with_no_helm_charts.helm, [])
+
+    def test_no_helm_charts_null(self):
+        """Test a product that has None under the 'helm' key returns an empty list."""
+        product_with_no_helm_charts = InstalledProductVersion(
+            'sat', '0.9.9', {'component_versions': {'helm': None}}
+        )
+        self.assertEqual(product_with_no_helm_charts.helm, [])
+
+    def test_no_helm_charts_empty_list(self):
+        """Test a product that has an empty list under the 'helm' key returns an empty list."""
+        product_with_no_helm_charts = InstalledProductVersion(
+            'sat', '0.9.9', {'component_versions': {'helm': []}}
+        )
+        self.assertEqual(product_with_no_helm_charts.helm, [])
+
+    def test_s3_artifacts(self):
+        """Test getting the s3 artifacts."""
+        expected_s3_artifacts = [('boot-images', 'PE/CPE-base.x86_64-2.0.squashfs'),
+                                 ('boot-images', 'PE/CPE-amd.x86_64-2.0.squashfs')]
+        product_with_s3_artifacts = InstalledProductVersion(
+            'cpe', '2.0.0', CPE_VERSION['2.0.0']
+        )
+        self.assertEqual(expected_s3_artifacts, product_with_s3_artifacts.s3_artifacts)
+
+    def test_no_s3_artifacts(self):
+        """Test a product that has an empty dictionary under the 's3' key returns an empty list."""
+        product_with_no_s3_artifacts = InstalledProductVersion(
+            'sat', '0.9.9', {'component_versions': {'s3': {}}}
+        )
+        self.assertEqual(product_with_no_s3_artifacts.s3_artifacts, [])
+
+    def test_no_s3_artifacts_null(self):
+        """Test a product that has None under the 's3' key returns an empty list."""
+        product_with_no_s3_artifacts = InstalledProductVersion(
+            'sat', '0.9.9', {'component_versions': {'s3': None}}
+        )
+        self.assertEqual(product_with_no_s3_artifacts.s3_artifacts, [])
+
+    def test_no_s3_artifacts_empty_list(self):
+        """Test a product that has an empty list under the 's3' key returns an empty list."""
+        product_with_no_s3_artifacts = InstalledProductVersion(
+            'sat', '0.9.9', {'component_versions': {'s3': []}}
+        )
+        self.assertEqual(product_with_no_s3_artifacts.s3_artifacts, [])
+
+    def test_loftsman_manifests(self):
+        """Test getting the Loftsman manifests."""
+        expected_loftsman_manifests = ['config-data/argo/loftsman/cos/2.0.0/manifests/cos-services.yaml']
+        product_with_loftsman_manifests = InstalledProductVersion(
+            'cos', '2.0.0', COS_VERSIONS['2.0.0']
+        )
+        self.assertEqual(expected_loftsman_manifests, product_with_loftsman_manifests.loftsman_manifests)
+
+    def test_no_loftsman_manifests_empty_list(self):
```

Comment:
Add a testcase, when the key is missing in the data and its being extracted from object.
Example: If in component version s3 is missing and in the `InstalledProductVersion` object use is able to get the value.